lib/guardian/category_guardian.rb

```diff
@@ -49,7 +49,6 @@ def can_post_in_category?(category)
     return false unless category
     return false if is_anonymous?
     return true if is_admin?
-    return true if !category.read_restricted
```

Comment:
We should make sure to update the specs around category guardian to account for this, in general changing a core guardian method like this is risky business since there are likely a lot of assumptions around it. What you've done is correct AFAICT since if you create a category with `everyone` group only having `see` then `read_restricted` is false.